lib/internal/bootstrap/node.js

```diff
@@ -67,12 +67,26 @@ if (ownsProcessState) {
 process.config = JSON.parse(internalBinding('native_module').config);
 
 const rawMethods = internalBinding('process_methods');
+
 // Set up methods and events on the process object for the main thread
 if (isMainThread) {
+  // Cache the working directory to prevent lots of lookups. If the working
+  // directory is changed by `chdir`, it'll be updated.
+  let cachedCwd = '';
+
   process.abort = rawMethods.abort;
   const wrapped = mainThreadSetup.wrapProcessMethods(rawMethods);
   process.umask = wrapped.umask;
-  process.chdir = wrapped.chdir;
+  process.chdir = (dir) => {
```

Comment:
Why not just do this in `mainThreadSetup.wrapProcessMethods(rawMethods)`? `bootstrap/node.js` focuses on what's attached to the globals, instead of how they are implemented.